### PR TITLE
Access log

### PR DESCRIPTION
**Pull request by plates1**
Log every request to the access log

### CODE REVIEW

**Comment by reynol2** [comment]
Did we want to give the ability to disable this logging?

**Comment by malibuworkcrew** [comment]
@reynol2 We chatted about that, I'll probably throw it in in a future release (so we added a TODO) but it might need a little jimmying to get at the configs from that class. Going to merge this to get it out then make that change later.

File: src/main/scala/com/webtrends/harness/component/akkahttp/logging/AccessLog.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      val responseTimestamp: Long = System.currentTimeMillis()
+      val requestTimestamp: Long = bean.getValue[Long](TimeOfRequest).getOrElse(responseTimestamp)
+      val elapsedTime: Long = responseTimestamp - requestTimestamp
+      val now: String = DateTime(requestTimestamp).toIsoDateTimeString()
+      val method: String = bean.get(AkkaHttpBase.Method).map {
+        case m: HttpMethod => m.value
+        case m => m.toString
+      }.getOrElse("-")
+      val path: String = reconstructPath(bean)
+      /*
```
Comment (Kraagen) [inline src/main/scala/com/webtrends/harness/component/akkahttp/logging/AccessLog.scala:37]: Comment doesn't match implementation.

File: src/main/scala/com/webtrends/harness/component/akkahttp/logging/AccessLog.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import com.webtrends.harness.component.akkahttp.AkkaHttpBase.TimeOfRequest
+import com.webtrends.harness.logging.Logger
+
+trait AccessLog  {
+  this: BaseCommand =>
+
+  val accessLog = Logger("AccessLog")
+  def logAccess(bean: CommandBean, statusCode: Option[StatusCode] = None) {
+
+    def reconstructPath(bean: CommandBean): String = {
```
Comment (Kraagen) [inline src/main/scala/com/webtrends/harness/component/akkahttp/logging/AccessLog.scala:15]: Do we want to just pass through the path so we don't have to bother reconstructing it every time?

File: src/main/scala/com/webtrends/harness/component/akkahttp/logging/AccessLog.scala [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      val segments = path.split("/").filter(_.startsWith("$"))
+      segments.foldLeft(path){(p,s) => bean.getValue[String](s.drop(1)).map(value => p.replace(s, value)).getOrElse(p)}
+    }
+
+    // modify the logback.xml file to write the "AccessLog" entries to a file without all of the prefix information
+    //TODO add a config with an option to turn off logging
+    try {
+      val host: String = bean.getValue[Map[String,String]](AkkaHttpBase.RequestHeaders).flatMap(rh => rh.get("host")).getOrElse("-")
+      // the userId may be either a String or an Int
+      val user: String = bean.get("userId").map(id => id.toString).getOrElse("-")
```
Comment (Kraagen) [inline src/main/scala/com/webtrends/harness/component/akkahttp/logging/AccessLog.scala:26]: Do we want to use this or the idcs user id from the header?

File: src/main/scala/com/webtrends/harness/component/akkahttp/logging/AccessLog.scala [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      val status:String = statusCode.map(sc => sc.intValue.toString).getOrElse("-")
+      val responseTimestamp: Long = System.currentTimeMillis()
+      val requestTimestamp: Long = bean.getValue[Long](TimeOfRequest).getOrElse(responseTimestamp)
+      val elapsedTime: Long = responseTimestamp - requestTimestamp
+      val now: String = DateTime(requestTimestamp).toIsoDateTimeString()
```
Comment (Kraagen) [inline src/main/scala/com/webtrends/harness/component/akkahttp/logging/AccessLog.scala:31]: Why call the variable now when it is the request time, not now?

**Pull request merged**